### PR TITLE
rpc/jsonrpc/server: return an error in WriteRPCResponseHTTP(Error)

### LINKED ISSUE #5529: RPC keeps panic when there are RPC requests
**Tendermint version** (use `tendermint version` or `git rev-parse --verify HEAD` if installed from source):
v0.33.7

**ABCI app** (name for built-in, URL for self-written if it's publicly available):
[Kava-4 ](https://github.com/Kava-Labs/kava/tree/v0.11.0) with Cosmos SDK `v0.39.1`.

**Environment**
- **OS** (e.g. from /etc/os-release): Ubuntu 20.04 LTS
- **Install tools**: go module
- **Others**:

**What happened**:
Node continuously having
```
Panic during RPC panic recovery: write tcp 127.0.0.1:26657->127.0.0.1:52708: i/o timeout
```
And not syncing with new blocks.

**What you expected to happen**:
The node would panic during RPC panic, stop the connection and continue syncing.

**Have you tried the latest version**: yes/no
no

**How to reproduce it** (as minimally and precisely as possible):

**Logs (paste a small part showing an error (< 10 lines) or link a pastebin, gist, etc. containing more of the log file)**:
https://pastebin.com/Q2ciBsaf

**Config (you can paste only the changes you've made)**:

**node command runtime flags**:

**Please provide the output from the `http://<ip>:<port>/dump_consensus_state` RPC endpoint for consensus bugs**

**Anything else we need to know**:
The RPC service is behind reverse proxy.

### PR DESCRIPTION
**Pull request by melekes**
instead of panicking
Closes #5529

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]
🔥

File: rpc/jsonrpc/server/http_server.go [hunk 1]
```diff
@@ -205,14 +202,12 @@ func RecoverAndLogHandler(handler http.Handler, logger log.Logger) http.Handler
...
 
-					logger.Error(
-						"Panic in RPC HTTP handler", "err", e, "stack",
-						string(debug.Stack()),
-					)
-					WriteRPCResponseHTTPError(
-						rww,
-						types.RPCInternalError(types.JSONRPCIntID(-1), err),
-					)
+					logger.Error("Panic in RPC HTTP handler", "err", e, "stack", string(debug.Stack()))
```
Comment (alexanderbez) [inline rpc/jsonrpc/server/http_server.go:205]:
```suggestion
					logger.Error("panic in RPC HTTP handler", "err", e, "stack", string(debug.Stack()))
```

**Review by alexanderbez — APPROVED** [review]
🎉

**Comment by michaelfig** [comment]
Am I correct in thinking this will fix my validator going MIA for a brief period (a block or two) with repeating `Panic during RPC panic recovery`?  Here is the log excerpt https://gist.github.com/michaelfig/40810a67c68c6991a070c7b06f29712b

Please LMK if that's encompassed in this PR or if I should file a new issue.  Thanks!

**Pull request merged**